src/core/eme/sessions_set/in_memory.ts

```diff
@@ -106,8 +111,17 @@ export default class InMemorySessionsSet extends SessionSet<ISessionData> {
    * @param {string} sessionId
    * @returns {MediaKeySession|null}
    */
-  deleteById(sessionId : string) : IMediaKeySession|MediaKeySession|null {
-    const entry = this.find((e) => e.session.sessionId === sessionId);
+  deleteByInitData(
+    initData : Uint8Array,
+    initDataType: string
+  ) : IMediaKeySession|MediaKeySession|null {
+    const hash = hashInitData(initData);
+    const entry = this.find((e) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update this function so that its implementation is not identical to the one on line 73. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=187&issues=AWLjaM94jr7wspRo8NBL&open=AWLjaM94jr7wspRo8NBL)

[ISSUE_KEY]: AWLjaM94jr7wspRo8NBL